language/go/config.go

```diff
@@ -236,6 +243,69 @@ func (f tagsFlag) String() string {
 	return ""
 }
 
+type namingConventionFlag struct {
+	nc *namingConvention
+}
+
+func (f namingConventionFlag) Set(value string) error {
+	if nc, err := namingConventionFromString(value); err != nil {
+		return err
+	} else {
+		*f.nc = nc
+		return nil
+	}
+}
+
+func (f *namingConventionFlag) String() string {
+	if f == nil || f.nc == nil {
+		return "naming_convention"
+	}
+	return f.nc.String()
+}
+
+// namingConvention determines how go targets are named.
+type namingConvention int
+
+const (
+	// 'go_default_library' and 'go_default_test'
+	goDefaultLibraryNamingConvention = iota
+
+	// For an import path that ends with foo, the go_library rules target is
+	// named 'foo', the go_test is named 'foo_test'.
+	// For a main package, the go_binary takes the 'foo' name, the library
+	// is named 'foo_lib', and the go_test is named 'foo_test'.
+	importNamingConvention
+
+	// Same as importNamingConvention, but generate alias rules for libraries that have
+	// the legacy 'go_default_library' name.
+	importAliasNamingConvention
+)
+
+func (nc namingConvention) String() string {
+	switch nc {
+	case goDefaultLibraryNamingConvention:
+		return "go_default_library"
+	case importNamingConvention:
+		return "import"
+	case importAliasNamingConvention:
+		return "import_alias"
+	}
+	return ""
+}
+
+func namingConventionFromString(s string) (namingConvention, error) {
+	switch s {
+	case "go_default_library":
+		return goDefaultLibraryNamingConvention, nil
+	case "import":
+		return importNamingConvention, nil
+	case "import_alias":
+		return importAliasNamingConvention, nil
+	default:
+		return goDefaultLibraryNamingConvention, fmt.Errorf("unknown naming convention %q", s)
```

Comment:
The default should be `goDefaultLibraryNamingConvention` to be backward compatible. Plus, the updated `README.md` says: "By default, library targets are named  ``go_default_library`` and test targets ``go_default_test``."